crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2532,6 +2572,17 @@ impl<'db> TypeInferenceBuilder<'db> {
                 .map(Type::IntLiteral)
                 .unwrap_or_else(|| KnownClass::Int.to_instance(self.db)),
 
+            (Type::IntLiteral(n), Type::IntLiteral(m), ast::Operator::Pow) => {
```

Comment:
Let's add a few mdtests for `Pow` operator on int literals in `binary/integers.md`? (One for each possible case here: ok literal result, too large `m`, too large result.)